internal/auditlog/https_writer_test.go

```diff
@@ -11,40 +11,45 @@ import (
 	"io"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 
 	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
 	"github.com/corazawaf/coraza/v3/types"
 )
 
+var sampleHttpsAuditLog = &Log{
+
+	Transaction_: Transaction{
+		ID_: "test123",
+	},
+	Messages_: []plugintypes.AuditLogMessage{
+		Message{
+			Data_: &MessageData{
+				ID_:  100,
+				Raw_: "SecAction \"id:100\"",
+			},
+		},
+	},
+}
+
 func TestHTTPSAuditLog(t *testing.T) {
 	writer := &httpsWriter{}
 	formatter := nativeFormatter
 	pts, err := types.ParseAuditLogParts("ABCDEZ")
 	if err != nil {
 		t.Fatal(err)
 	}
-	al := &Log{
-		Parts_: pts,
-
-		Transaction_: Transaction{
-			ID_: "test123",
-		},
-		Messages_: []plugintypes.AuditLogMessage{
-			Message{
-				Data_: &MessageData{
-					ID_:  100,
-					Raw_: "SecAction \"id:100\"",
-				},
-			},
-		},
-	}
+	sampleHttpsAuditLog.Parts_ = pts
 	// we create a test http server
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 		if r.ContentLength == 0 {
 			t.Fatal("ContentLength is 0")
 		}
+		if ct := r.Header.Get("Content-Type"); !strings.HasPrefix(ct, "application/octet-stream") {
```

Comment:
Is HasPrefix accurate? Can't we do direct comparison equality?

Reply:
Im most cases yes, but at some point it would be right to use application/json;encoding=utf-8, like the automatic mime detection does